custom_methods.rb

```diff
@@ -0,0 +1,148 @@
+# rubocop:disable Style/CaseEquality
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def change_self
+    return Array self if self.class == Range
+
+    self
+  end
+
+  def my_each
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i]
+      i += 1
+    end
+    my_array
+  end
+
+  def my_each_with_index
+    my_array = change_self
+    return to_enum unless block_given?
+
+    i = 0
+    while i < my_array.length
+      yield my_array[i], i
+      i += 1
+    end
+    my_array
+  end
+
+  def my_select
+    my_array = change_self
+    return to_enum unless block_given?
+
+    arr = []
+    my_array.my_each { |x| arr.push(x) if yield x }
+    arr
+  end
+
+  def my_all?(word = nil)
+    my_array = change_self
+    if block_given?
+      my_array.my_each { |item| return false unless yield item }
+      true
+    elsif word
+      my_array.my_each { |item| return false unless n === item }
+      true
+    else
+      my_all? { |item| item }
+    end
+  end
+
+  def my_any?(regex = false)
```

Comment:
* [ ] There should be a working `my_any?`  identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
compare answers in each of theexamples below
``` p %w[ant bear cat].my_any?(/d/)``` != ```%w[ant bear cat].any?(/d/)```
``` p [nil, true, 99].my_any?(Integer)```  != ```[nil, true, 99].any?(Integer)```